lib/manager/pipenv/artifacts.ts

```diff
@@ -72,7 +103,7 @@ export async function updateArtifacts({
         image: 'renovate/python',
         tagConstraint,
         tagScheme: 'pep440',
-        preCommands: ['pip install --user pipenv'],
+        preCommands: ['pip install --user pipenv' + pipenvConstraint],
```

Comment:
```suggestion
        preCommands: [`pip install --user pipenv${quote(pipenvConstraint)}`],
```

Reply:
This will require `import { quote } from 'shlex';` be added at the top too

Reply:
@rarkins I'm not sure if this will work, if the version is quoted, maybe the full string (`"pipenv==1.2.3"`) needs to be quoted. 🤔

Reply:
Yes, good point, it needs a refactor to quote the full string as you suggest

Reply:
Done